.github/workflows/formatter.yml

```diff
@@ -0,0 +1,34 @@
+name: Code Formatter
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  formatter:
+    name: "Format Codebase"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 2  # In PR, has extra merge commit: ^1 = PR, ^2 = base
+
+      - name: Format with NPH
```

Comment:
This isn't to format the code, it should already be formatted. This is to make sure the code has been formatted.

Reply:
I think you might have misread the repo's documentation. In order to do that you need to provide the `--check <file>` flag, otherwise it does indeed format the files.

Besides this, are you aware if we are actually enforcing `nph` formatting before committing? Either "manually" or via a pre-commit hook?

Reply:
This is a job that is gonna fail if the code hasn't been formatted and therefore the PR can't be merged. It won't commit any change it might produce. Maybe it's possible to use check indeed, but it is already working as intended and doesn't require changes.

Reply:
Ah, I hadn't noticed the `git diff --exit-code` line. That explains it.